lib/ansible/plugins/__init__.py

```diff
@@ -141,3 +141,6 @@ def _no_options(self, *args, **kwargs):
     @property
     def j2_function(self):
         return self._function
+
+
+PLUGIN_INSTANCE_CACHE = {}  # type: dict[str, dict[str, AnsiblePlugin]]
```

Comment:
this cache won't be updated correctly from forks, as the data will be lost once the fork is completed